### PR TITLE
Pin rust to <1.49 on osx-64

### PR DESCRIPTION
**Pull request by mbargull**
due to min deployment target req,
cf. https://github.com/conda-forge/rust-feedstock/issues/73

cc @conda-forge/rust, @isuruf

### CODE REVIEW

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -615,6 +615,9 @@ root_base:
 ruby:
   - 2.5
   - 2.6
+# Pin rust to <1.49 on osx-64 due to min deployment target req, cf. https://github.com/conda-forge/rust-feedstock/issues/73
+rust:     # [osx and x86_64]
+  - 1.48  # [osx and x86_64]
```
Comment (mbargull) [inline recipe/conda_build_config.yaml:620]: In reality we'd only need `<1.49`. But I'm not sure how/if our tooling supports pinning expressions apart from `X.Y.*`.

**Review by isuruf — CHANGES_REQUESTED** [review]
You also need to pin `rust_compiler_version`

**Comment by mbargull** [comment]
> You also need to pin `rust_compiler_version`

Right, I forgot that pinnings aren't applied to implicit deps, thanks, fixed.

File: recipe/conda_build_config.yaml [hunk 2]
```diff
@@ -82,6 +82,9 @@ target_gobin:
 # Rust Compiler Options
 rust_compiler:
   - rust
+# Pin rust to <1.49 on osx-64 due to min deployment target req, cf. https://github.com/conda-forge/rust-feedstock/issues/73
+rust_compiler_version:  # [osx and x86_64]
+  - 1.48                # [osx and x86_64]
```
Comment (isuruf) [inline recipe/conda_build_config.yaml:87]: There's no 1.48 AFAIK. 1.46 maybe

Comment (mbargull) [reply recipe/conda_build_config.yaml]: I didn't notice those were out of sync. Fixed, thanks.

**Pull request merged**